include/swupd.h

```diff
@@ -258,6 +258,10 @@ extern int get_dirfd_path(const char *fullname);
 extern int verify_fix_path(char *targetpath, struct manifest *manifest);
 extern void set_local_download(void);
 extern struct list *files_from_bundles(struct list *bundles);
+extern int system_argv(char *const argv[]);
+extern int system_argv_fd(char *const argv[], int newstdin, int newstdout, int newstderr);
+extern int system_argv_pipe(char *const argvp1[], int stdinp1, int stderrp1,
```

Comment:
these signatures are easy to get wrong, what if you use a structure? Then at least it could use member initializers and be cleaner:
```c
struct system_argv_pipe_args {
    struct {
       char *const argv[];
       int stdin;
       int stderr;
   } p1;
 struct {
       char *const argv[];
       int stdout;
       int stderr;
   } p2;   
};

#define system_argv_fd(...) system_argv_fd_impl((const struct system_argv_pipe_args){__VA_ARGS});

system_argv_fd(.p1 = {.stdin = STDIN_FILENO, argv = ...}, .p2 = ...);
```